packages/nuxt/src/core/nuxt.ts

```diff
@@ -663,6 +663,11 @@ async function initNuxt (nuxt: Nuxt) {
       filePath: resolve(nuxt.options.appDir, 'components/nuxt-island'),
     })
 
+    addServerHandler({
+      route: '/__nuxt_island/**',
+      handler: resolve(distDir, 'core/runtime/nitro/handlers/__nuxt_island'),
+    })
```

Comment:
we might be able to add this conditionally - e.g. not add it if the user has disabled `experimental.componentIslands` entirely.